src/tree/tree.rs

```diff
@@ -125,4 +125,20 @@ where
             _ => Err(CanonError::NotFound),
         }
     }
+
+    /// Provides an iterator over the leaves of the tree which have been previously annotated via a custom `Walker` passed
+    /// as argument.
+    ///
+    /// # Note
+    /// This is only useful if annotate the tree is going to make the iteration perform sub-linearly.
+    pub fn annotated_iter_walk(
+        &self,
+        walker: impl Walker<NStack<L, A>, A>,
```

Comment:
i would go for a <W> where W: Walker on this, using impl in argument position is a bit controversial.

Reply:
Makes sense. Changed!